processor/src/test/java/org/mapstruct/ap/internal/util/StringsTest.java

```diff
@@ -0,0 +1,99 @@
+
+        assertThat( Strings.capitalize( null ) ).isNull();
+        assertThat( Strings.capitalize( "c" ) ).isEqualTo( "C" );
+        assertThat( Strings.capitalize( "capitalize" ) ).isEqualTo( "Capitalize" );
+        assertThat( Strings.capitalize( "AlreadyCapitalized" ) ).isEqualTo( "AlreadyCapitalized" );
+        assertThat( Strings.capitalize( "notCapitalized" ) ).isEqualTo( "NotCapitalized" );
+
+    }
+
+    @Test
+    public void testDecapitalize() throws Exception {
+        assertThat( Strings.decapitalize( null ) ).isNull();
+        assertThat( Strings.decapitalize( "c" ) ).isEqualTo( "c" );
+        assertThat( Strings.decapitalize( "capitalize" ) ).isEqualTo( "capitalize" );
+        assertThat( Strings.decapitalize( "AlreadyCapitalized" ) ).isEqualTo( "alreadyCapitalized" );
+        assertThat( Strings.decapitalize( "notCapitalized" ) ).isEqualTo( "notCapitalized" );
+    }
+
+    @Test
+    public void testJoin() throws Exception {
+        assertThat( Strings.join( new ArrayList<String>(), "-" ) ).isEqualTo( "" );
+        assertThat( Strings.join( Arrays.asList( "Hello", "World" ), "-" ) ).isEqualTo( "Hello-World" );
+        assertThat( Strings.join( Arrays.asList( "Hello" ), "-" ) ).isEqualTo( "Hello" );
+    }
+
+    @Test
+    public void testJoinAndCamelize() throws Exception {
+        assertThat( Strings.joinAndCamelize( new ArrayList<String>() ) ).isEqualTo( "" );
+        assertThat( Strings.joinAndCamelize( Arrays.asList( "Hello", "World" ) ) ).isEqualTo( "HelloWorld" );
+        assertThat( Strings.joinAndCamelize( Arrays.asList( "Hello", "world" ) ) ).isEqualTo( "HelloWorld" );
+        assertThat( Strings.joinAndCamelize( Arrays.asList( "hello", "world" ) ) ).isEqualTo( "helloWorld" );
+    }
+
+    @Test
+    public void testIsEmpty() throws Exception {
+        assertThat( Strings.isEmpty( null ) ).isTrue();
+        assertThat( Strings.isEmpty( "" ) ).isTrue();
+        assertThat( Strings.isEmpty( "not empty" ) ).isFalse();
+
+    }
+
+    @Test
+    public void testGetSaveVariableNameWithArrayExistingVariables() throws Exception {
+        assertThat( Strings.getSaveVariableName( "int[]" ) ).isEqualTo( "intArray" );
+        assertThat( Strings.getSaveVariableName( "Extends" ) ).isEqualTo( "extends_" );
+        assertThat( Strings.getSaveVariableName( "prop", "prop", "prop_" ) ).isEqualTo( "prop__" );
+    }
+
+    @Test
+    public void testGetSaveVariableNameWithCollection() throws Exception {
+        assertThat( Strings.getSaveVariableName( "int[]", new ArrayList<String>() ) ).isEqualTo( "intArray" );
+        assertThat( Strings.getSaveVariableName( "Extends", new ArrayList<String>() ) ).isEqualTo( "extends_" );
+        assertThat( Strings.getSaveVariableName( "prop", Arrays.asList( "prop", "prop_" ) ) ).isEqualTo( "prop__" );
+        assertThat( Strings.getSaveVariableName( "prop.font", Arrays.asList( "propFont", "propFont_" ) ) )
+            .isEqualTo( "propFont__" );
+
+
+    }
```

Comment:
Yes I went for this branch, as this can be a big problem for 1.1. There is no nice workaround for this